_posts/2020-10-16-how-we-estimate-rails-application-size.md

```diff
@@ -0,0 +1,176 @@
+|               sass-rails | 31         | 5              |
+|              rspec-rails | 28         | 7              |
+|              web-console | 25         | 4              |
+|             dotenv-rails | 24         | 2              |
+|        factory_bot_rails | 24         | 2              |
+|                webpacker | 24         | 3              |
+|              tracker_api | 19         | 9              |
+| rails-controller-testing | 19         | 3              |
+|          omniauth-github | 10         | 2              |
+|                 capybara | 9          | 7              |
+|                 jbuilder | 7          | 1              |
+|    spring-watcher-listen | 6          | 2              |
+|            rspec-sidekiq | 6          | 2              |
+|                  webmock | 5          | 3              |
+|                  octokit | 5          | 2              |
+|               webdrivers | 5          | 3              |
+|                  codecov | 4          | 2              |
+|                   listen | 4          | 3              |
+|                  sidekiq | 3          | 3              |
+|                simplecov | 2          | 2              |
+|       selenium-webdriver | 2          | 2              |
+|                     puma | 1          | 1              |
+|               turbolinks | 1          | 1              |
+|                 bootsnap | 1          | 1              |
+|                      vcr | 0          | 0              |
+|                   spring | 0          | 0              |
+|         database_cleaner | 0          | 0              |
+|                   byebug | 0          | 0              |
+|                  lockbox | 0          | 0              |
+|                       pg | 0          | 0              |
+|              tzinfo-data | 0          | 0              |
++--------------------------|------------|----------------+
+
+      Declared Gems   33
+         Total Gems   129
+  Unpinned Versions   23
+        Github Refs   0
+```
+
+Now we know that our application declares 33 gems in its `Gemfile`, but the
+total gems is actually 129 gems.
+
+**Total gems** is an important metric because dependencies are constantly
+changing which might be an issue if you want to upgrade from Rails 6.0 to
+6.1.
+
+### Codebase (Lines of Code)
+
+In order to calculate the lines of code in a codebase, I prefer
+[`rails_stats`](https://github.com/bleonard/rails_stats) which enhances Rails's
+native `rake stats`.
+
+One of the things I like about it is that `rails_stats` doesn't need to load the
+Rails environment in order to calculate all statistics. Sometimes that can be
+annoying, especially when you are trying to size up a really, really old
+application which requires a very specific set of dependencies (think a very
+specific version of Ruby, Rails, Bundler, and Rubygems)
+
+You will need a `Rakefile` in the directory where you call `rake stats` and you
```

Comment:
Maybe here you can explain why this is required? I understand it's because you are running the rails_stats gem standalone and from outside a Rails app directory, but it's not clear at first glance.

Also in the `bundler-stats` example you showed the installation command, maybe it should be the same here for consistency

Reply:
Good point 👍